File: Scripts/enums/vehicle_enums.lua
local VehicleClass = {
    Compact = 0,
    Sedan = 1,
    SUV = 2,
    Coupe = 3,
    Muscle = 4,
    SportsClassic = 5,
    Sport = 6,
    Super = 7,
    Motorcycle = 8,
    Offroad = 9,
    Industrial = 10,
    Utility = 11,
    Van = 12,
    Cycle = 13,
    Boat = 14,
    Helicopter = 15,
    Plane = 16,
    Service = 17,
    Emergency = 18,
    Military = 19,
    Commercial = 20,
    Train = 21
}

local VehicleWindow = {
    FrontLeftWindow = 0,
    FrontRightWindow = 1,
    BackLeftWindow = 2,
    BackRightWindow = 3,
    Last = 4
}

local CargobobHook = {
    Hook = 0,
    Magnet = 1
}

local VehicleLockStatus = {
    None = 0,
    Unlocked = 1,
    Locked = 2,
    LockedForPlayer = 3,
    StickPlayerInside = 4,
    CanBeBrokenInto = 7,
    CanBeBrokenIntoPersist = 8,
    CannotBeTriedToEnter = 10
}

local VehicleMod = {
    Spoiler = 0,
    FrontBumper = 1,
    RearBumper = 2,
    SideSkirt = 3,
    Exhaust = 4,
    Frame = 5,
    Grille = 6,
    Hood = 7,
    Fender = 8,
    RightFender = 9,
    Roof = 10,
    Engine = 11,
    Brakes = 12,
    Transmission = 13,
    Horns = 14,
    Suspension = 15,
    Armor = 16,
    Unknown17 = 17,
    Turbo = 18,
    Unknown19 = 19,
    TireSmoke = 20,
    Unknown21 = 21,
    XenonHeadlights = 22,
    FrontWheels = 23,
    BackWheels = 24, -- Only for motorcycles
    Plateholder = 25,
    VanityPlate = 26,
    TrimDesign = 27,
    Ornament = 28,
    Dash = 29,
    DialDesign = 30,
    SpeakersDoor = 31,
    LeatherSeats = 32,
    SteeringWheel = 33,
    ColumnShifterLever = 34,
    Plaque = 35,
    Speakers = 36,
    Trunk = 37,
    Hydraulics = 38,
    EngineBlock = 39,
    AirFilter = 40,
    Struts = 41,
    ArchCover = 42,
    Aerials = 43,
    Trim = 44,
    Tank = 45,
    Windows = 46,
    Unknown47 = 47,
    Livery = 48
}

local NumberPlateType = {
    FrontAndRearPlates = 0,
    FrontPlate = 1,
    RearPlate = 2,
    None = 3
}

local WheelType = {
    Stock = -1,
    Sport = 0,
    Muscle = 1,
    Lowrider = 2,
    SUV = 3,
    Offroad = 4,
    Tuner = 5,
    BikeWheels = 6,
    HighEnd = 7,
    Bennys = 8,
    BennysBespoke = 9
}

local WindowTint = {
    None = 0,
    Light = 3,
    Medium = 2,
    Dark = 4,
    Limo = 1,
    Green = 6
}

return {
    VehicleClass = VehicleClass,
    VehicleWindow = VehicleWindow,
    CargobobHook = CargobobHook,
    VehicleLockStatus = VehicleLockStatus,
    VehicleMod = VehicleMod,
    NumberPlateType = NumberPlateType,
    WheelType = WheelType,
    WindowTint = WindowTint
}

File: Scripts/enums/vehicle_enums.py
class VehicleClass:
    COMPACT = 0
    SEDAN = 1
    SUV = 2
    COUPE = 3
    MUSCLE = 4
    SPORTS_CLASSIC = 5
    SPORT = 6
    SUPER = 7
    MOTORCYCLE = 8
    OFFROAD = 9
    INDUSTRIAL = 10
    UTILITY = 11
    VAN = 12
    CYCLE = 13
    BOAT = 14
    HELICOPTER = 15
    PLANE = 16
    SERVICE = 17
    EMERGENCY = 18
    MILITARY = 19
    COMMERCIAL = 20
    TRAIN = 21

class VehicleWindow:
    FRONT_LEFT_WINDOW = 0
    FRONT_RIGHT_WINDOW = 1
    BACK_LEFT_WINDOW = 2
    BACK_RIGHT_WINDOW = 3
    LAST = 4

class CargobobHook:
    HOOK = 0
    MAGNET = 1

class VehicleLockStatus:
    NONE = 0
    UNLOCKED = 1
    LOCKED = 2
    LOCKED_FOR_PLAYER = 3
    STICK_PLAYER_INSIDE = 4
    CAN_BE_BROKEN_INTO = 7
    CAN_BE_BROKEN_INTO_PERSIST = 8
    CANNOT_BE_TRIED_TO_ENTER = 10

class VehicleMod:
    SPOILER = 0
    FRONT_BUMPER = 1
    REAR_BUMPER = 2
    SIDE_SKIRT = 3
    EXHAUST = 4
    FRAME = 5
    GRILLE = 6
    HOOD = 7
    FENDER = 8
    RIGHT_FENDER = 9
    ROOF = 10
    ENGINE = 11
    BRAKES = 12
    TRANSMISSION = 13
    HORNS = 14
    SUSPENSION = 15
    ARMOR = 16
    UNKNOWN_17 = 17
    TURBO = 18
    UNKNOWN_19 = 19
    TIRE_SMOKE = 20
    UNKNOWN_21 = 21
    XENON_HEADLIGHTS = 22
    FRONT_WHEELS = 23
    BACK_WHEELS = 24 # Only for motorcycles
    PLATEHOLDER = 25
    VANITY_PLATE = 26
    TRIM_DESIGN = 27
    ORNAMENT = 28
    DASH = 29
    DIAL_DESIGN = 30
    SPEAKERS_DOOR = 31
    LEATHER_SEATS = 32
    STEERING_WHEEL = 33
    COLUMN_SHIFTER_LEVER = 34
    PLAQUE = 35
    SPEAKERS = 36
    TRUNK = 37
    HYDRAULICS = 38
    ENGINE_BLOCK = 39
    AIR_FILTER = 40
    STRUTS = 41
    ARCH_COVER = 42
    AERIALS = 43
    TRIM = 44
    TANK = 45
    WINDOWS = 46
    UNKNOWN_47 = 47
    LIVERY = 48

class NumberPlateType:
    FRONT_AND_REAR_PLATES = 0
    FRONT_PLATE = 1
    REAR_PLATE = 2
    NONE = 3

class WheelType:
    STOCK = -1
    SPORT = 0
    MUSCLE = 1
    LOWRIDER = 2
    SUV = 3
    OFFROAD = 4
    TUNER = 5
    BIKE_WHEELS = 6
    HIGH_END = 7
    BENNYS = 8
    BENNYS_BESPOKE = 9

class WindowTint:
    NONE = 0
    LIGHT = 3
    MEDIUM = 2
    DARK = 4
    LIMO = 1
    GREEN = 6
